File: common/util/TimerEvent.cpp

#include "TimerEvent.h"
#include "seda/EventType.h"

namespace zonda
{
namespace common
{

TimerEvent::TimerEvent(int id, int64_t param):
	m_id(id), m_param(param)
{
}

int TimerEvent::get_type() const
{
	return EventType::TIMER_EVENT;
}

int TimerEvent::get_id() const
{
	return m_id;
}

int64_t TimerEvent::get_param() const
{
	return m_param;
}



}//namespace common
}//namespace zonda

File: common_java/src/main/java/com/alibaba/platform/zonda/common/util/TimerEvent.java
package com.alibaba.platform.zonda.common.util;
import com.alibaba.platform.zonda.common.seda.*;


public class TimerEvent implements IEvent
{
    private int m_id;
    private long m_param;

    public TimerEvent(int id, long param)
    {
        m_id = id;
        m_param = param;
    }
    public int getType()
    {
        return EventType.TIMER_EVENT;
    }
    public int getId()
    {
        return m_id;
    }
    public long getParam()
    {
        return m_param;
    }
};